iceoryx_posh/include/iceoryx_posh/popo/sample.hpp

```diff
@@ -66,22 +67,29 @@ struct SamplePrivateData<const T>
 /// @brief The Sample class is a mutable abstraction over types which are written to loaned shared memory.
 /// These samples are publishable to the iceoryx system.
 ///
-template <typename T>
+template <typename T, typename H = typename cxx::add_const_conditionally<mepoo::NoCustomHeader, T>::type>
 class Sample
 {
+    static_assert(std::is_const<T>::value == std::is_const<H>::value,
```

Comment:
According to another `static_assert` they both have to be non const?! Maybe we should centralize this in some kind of concept or at least using alias?!

Or did I miss something?

Reply:
You are referring to the `static_assert` in the publisher, right? That's correct. This assert just checks if the types are equal in their const qualifiers, so the assertion in the publisher is still valid.